Update archiveloop: Log temperatures

THX723 opened the pull request:
Add temperature logging to provide the thermo profile of the SBC as it operates. It is event triggered so as not to cluttered the log. Triggers 1) shortly after boot-up, 2) after each snapshot (~1hr by default), and 3) after archiving.

[SampleLog.txt](https://github.com/user-attachments/files/20914940/SampleLog.txt)

THX723 committed:
Update archiveloop: Log temperatures

THX723 committed:
Update archiveloop

THX723 reviewed (commented):
add quotes for log string.

THX723 committed:
Update archiveloop

Tweaked temperature measurement points for file transfers -- moved under archive_clips(). Added another sampling after music file transfer (this process can sometime take a long time, so a thermo loading point).

THX723 committed:
Update archiveloop

THX723 commented:
Tweaked temperature measurement points -- now moved under archive_clips(). Added another sampling point after music file transfer, as this process can sometime take a long time (thermo loading).

marcone commented on run/archiveloop:
```diff
@@ -687,6 +687,8 @@ function archive_clips () {
     log "Archiving failed."
   fi
 
+  log "$(vcgencmd measure_temp)"
```
`vcgencmd` is Raspberry Pi/OS specific, so should be conditional on running on the right hardware/software.
You might consider reading /sys/class/thermal/thermal_zone0/temp instead, which should work pretty much everywhere.

THX723 replied:
This is a good suggestion. Will change.

marcone reviewed (commented):
This doesn't seem super useful in general, and since it logs at every snapshot, this could be every 8 minutes or so. I would prefer to gate the logging on sudden temperature changes and/or only log when it goes over/under certain thresholds.
Note that you can also see the temperature in the web interface.

THX723 commented:
> This doesn't seem super useful in general, and since it logs at every snapshot, this could be every 8 minutes or so. I would prefer to gate the logging on sudden temperature changes and/or only log when it goes over/under certain thresholds. Note that you can also see the temperature in the web interface.

Peak operating temp will always happen right after a hard archive/transfer session, so that's the ideal time to capture. The rationale behind also measuring after snapshots is to get a glimpse of how hot/cool it gets throughout the day (under the sun perhaps), whether driving or sentry-ing. I doubt anyone will be staring at their Web UI during those times. While I like the ideal of threshold logging in general, I think it's just as useful to know how cool it was operating. The logging is lightweight enough that it shouldn't bother anyone looking through the log.

THX723 committed:
Update archiveloop

Made platform agnostic by reading temperature from /sys/class/thermal/thermal_zone0/temp

THX723 committed:
Update archiveloop

THX723 commented:
@marcone fixed.
